### ISSUE TITLE
ztest: vdev.c:1746: Assertion `scn->scn_phys.scn_min_txg <= vdev_dtl_min(vd)

### ISSUE THREAD

**Issue by behlendorf** [zfsonlinux/zfs#2302]
A long standing issue which ztest hits occasionally.

```
5 vdevs, 7 datasets, 23 threads, 600 seconds...
Pass   1,  SIGKILL,   0 ENOSPC, 15.4% of  270M used,   8% done,    9m11s to go
Pass   2,  SIGKILL,   0 ENOSPC, 21.7% of  270M used,  18% done,    8m11s to go
Pass   3,  SIGKILL,   0 ENOSPC, 22.1% of  270M used,  20% done,    8m00s to go
Pass   4,  SIGKILL,   0 ENOSPC, 14.2% of  508M used,  30% done,    6m58s to go
Pass   5,  SIGKILL,   0 ENOSPC, 16.1% of  508M used,  36% done,    6m23s to go
Pass   6,  SIGKILL,   0 ENOSPC, 17.5% of  508M used,  45% done,    5m27s to go
Pass   7,  SIGKILL,   0 ENOSPC, 17.1% of  508M used,  47% done,    5m16s to go
Pass   8,  SIGKILL,   0 ENOSPC, 16.9% of  508M used,  52% done,    4m48s to go
Pass   9, Complete,   0 ENOSPC, 15.9% of  576M used,  65% done,    3m31s to go
Pass  10,  SIGKILL,   0 ENOSPC, 17.6% of  576M used,  75% done,    2m29s to go
ztest: ../../module/zfs/vdev.c:1746: Assertion `scn->scn_phys.scn_min_txg <= vdev_dtl_min(vd) (0x4e3 <= 0x3)' failed.
child died with signal 6
```

**Comment by ryao** [comment]
This might be fixed by #3172. However, we will need to stress test ztest and not see this for an extended period of time before there could be any certainty.

**Comment by behlendorf** [comment]
Likely fixed by #4790.

**Comment by behlendorf** [comment]
Fixed by #4790.
